### PR TITLE
Overwrite port number for liveness/readiness probes

### PR DESCRIPTION
**Pull request by kylape**
Devs currently have to manually copy the port number from the
environment spec to their liveness/readiness probes in their app spec.
This would automatically update the probes with the correct port number
given an app uses the HTTPGet handler.

### CODE REVIEW

**Comment by psav** [comment]
This needs a little rethinking due to people using nonstandard ports for their checks, ie checks against metrics, should perhaps be gated behind a check for web port presence

**Comment by kylape** [comment]
Refined this PR.  Please refer to commit message for details.

File: config/manager/kustomization.yaml [hunk 1]
```diff
@@ -5,4 +5,4 @@ kind: Kustomization
 images:
 - name: controller
   newName: 127.0.0.1:5000/clowder
-  newTag: 7248ca5
+  newTag: 95777a1
```
Comment (psav) [inline config/manager/kustomization.yaml:8]: This needs removing

File: config/manifests/bases/clowder.clusterserviceversion.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ metadata:
   annotations:
     alm-examples: '[]'
     capabilities: Basic Install
-    operators.operatorframework.io/builder: operator-sdk-v1.0.1
+    operators.operatorframework.io/builder: operator-sdk-v1.0.0
```
Comment (psav) [inline config/manifests/bases/clowder.clusterserviceversion.yaml:7]: Same as above

File: controllers/cloud.redhat.com/makers/maker.go [hunk 3]
```diff
@@ -554,9 +620,16 @@ func initDeployment(app *crd.ClowdApp, env *crd.ClowdEnvironment, d *apps.Deploy
...
-	if deployment.Web {
+	if deployment.WebServices.Private.Enabled {
 		c.Ports = append(c.Ports, core.ContainerPort{
-			Name:          "web",
+			Name:          "private",
+			ContainerPort: env.Spec.Providers.Web.Port,
+		})
+	}
+
+	if deployment.WebServices.Public.Enabled {
```
Comment (psav) [inline controllers/cloud.redhat.com/makers/maker.go:630]: We need an or in here to keep backwards compat with `.Web` until it's gone

**Review by psav — CHANGES_REQUESTED** [review]

**Comment by jameswnl** [comment]
is this addressed already? do we still keep the port hardcoded in clowder  app template?

**Comment by bennyturns** [comment]
Yes, this is addressed in a different PR.

**Comment by bennyturns** [comment]
Actually let me check with pete and kyle, I am not 100% sure now

**Comment by BlakeHolifield** [comment]
Yeah @jameswnl this code is in master here: https://github.com/RedHatInsights/clowder/blob/master/controllers/cloud.redhat.com/providers/web/impl.go#L50

We currently don't set that value in the ephemeral or stage environments, so it will be default 10000. You can override and test on a local environment if you'd like.

**Pull request closed without merging**